### ISSUE TITLE
在embed_vehicleID_fc.py代码问题

### ISSUE THREAD

**Issue by daiguangzhao** [yanbai1993/Embedding-Network#13]
小姐姐你好~
感谢您和您团队的工作!
在embed_vehicleID_fc.py 105行-112行的代码,如下:
 for im in img:
            im = im.cuda()
            embd, _ = model(im)
            embd = embd.detach().cpu().numpy()
            embds.append(embd)
        embed = sum(embds) / len(embds)  # ??????

        embeddings.append(embed)
为什么要把一个mini-batch的特征进行均值化,10张图片的特征这样做之后,embed = sum(embds) / len(embds)这样做之后特征纬度不就改变了吗?

**Comment by yanbai1993** [comment]
对于图a，img是图a的各种crop的集合（比如对图a随机裁剪5次）。这段代码的操作是对于图a的几个样本的特征取均值。
如果你不进行crop操作，或其它处理，img仅有一张图像，所以这段代码处理之后，不改变img特征。

发自我的iPhone

在 2020年6月19日，21:42，daiguangzhao <notifications@github.com> 写道：

﻿

小姐姐你好~
感谢您和您团队的工作!
在embed_vehicleID_fc.py 105行-112行的代码,如下:
for im in img:
im = im.cuda()
embd, _ = model(im)
embd = embd.detach().cpu().numpy()
embds.append(embd)
embed = sum(embds) / len(embds) # ??????

    embeddings.append(embed)


为什么要把一个mini-batch的特征进行均值化,10张图片的特征这样做之后,embed = sum(embds) / len(embds)这样做之后特征纬度不就改变了吗?

**Comment by daiguangzhao** [comment]
感谢您的回复!
我入门不久,您好,关于您论文中提到的embedding以及embedding space主要体现在这份代码的哪个部分(最好是哪块区域);此外, 这样做的目的是什么?
感谢~
